Makefile

```diff
@@ -245,24 +245,28 @@ GIT_VERSION: force
+CRDS_CILIUM_V2 := ciliumnetworkpolicies \
+                  ciliumclusterwidenetworkpolicies \
+                  ciliumendpoints \
+                  ciliumidentities \
+                  ciliumnodes \
+                  ciliumexternalworkloads \
+                  ciliumlocalredirectpolicies \
+                  ciliumegressgatewaypolicies \
+                  ciliumenvoyconfigs \
+                  ciliumclusterwideenvoyconfigs
+CRDS_CILIUM_V2ALPHA1 := ciliumendpointslices \
+                        ciliumbgppeeringpolicies \
+                        ciliumloadbalancerippools \
+                        ciliumnodeconfigs
 manifests: ## Generate K8s manifests e.g. CRD, RBAC etc.
 	$(eval TMPDIR := $(shell mktemp -d))
-	$(QUIET)$(GO) run sigs.k8s.io/controller-tools/cmd/controller-gen $(CRD_OPTIONS) paths="$(PWD)/pkg/k8s/apis/cilium.io/v2;$(PWD)/pkg/k8s/apis/cilium.io/v2alpha1" output:crd:artifacts:config="$(TMPDIR)"
+	$(QUIET)$(GO) run sigs.k8s.io/controller-tools/cmd/controller-gen $(CRD_OPTIONS) paths=$(CRD_PATHS) output:crd:artifacts:config="$(TMPDIR)"
 	$(QUIET)$(GO) run ./tools/crdcheck "$(TMPDIR)"
-	mv ${TMPDIR}/cilium.io_ciliumnetworkpolicies.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumnetworkpolicies.yaml
-	mv ${TMPDIR}/cilium.io_ciliumclusterwidenetworkpolicies.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumclusterwidenetworkpolicies.yaml
-	mv ${TMPDIR}/cilium.io_ciliumendpoints.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumendpoints.yaml
-	mv ${TMPDIR}/cilium.io_ciliumidentities.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumidentities.yaml
-	mv ${TMPDIR}/cilium.io_ciliumnodes.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumnodes.yaml
-	mv ${TMPDIR}/cilium.io_ciliumexternalworkloads.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumexternalworkloads.yaml
-	mv ${TMPDIR}/cilium.io_ciliumlocalredirectpolicies.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumlocalredirectpolicies.yaml
-	mv ${TMPDIR}/cilium.io_ciliumegressgatewaypolicies.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumegressgatewaypolicies.yaml
-	mv ${TMPDIR}/cilium.io_ciliumendpointslices.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2alpha1/ciliumendpointslices.yaml
-	mv ${TMPDIR}/cilium.io_ciliumclusterwideenvoyconfigs.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumclusterwideenvoyconfigs.yaml
-	mv ${TMPDIR}/cilium.io_ciliumenvoyconfigs.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/ciliumenvoyconfigs.yaml
-	mv ${TMPDIR}/cilium.io_ciliumbgppeeringpolicies.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2alpha1/ciliumbgppeeringpolicies.yaml
-	mv ${TMPDIR}/cilium.io_ciliumloadbalancerippools.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2alpha1/ciliumloadbalancerippools.yaml
-	mv ${TMPDIR}/cilium.io_ciliumnodeconfigs.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2alpha1/ciliumnodeconfigs.yaml
+	for file in $(CRDS_CILIUM_V2); do mv ${TMPDIR}/cilium.io_$${file}.yaml ./pkg/k8s/apis/cilium.io/client/crds/v2/$${file}.yaml; done
```

Comment:
we could get rid of the crd-to-version mapping with the following command which extracts the necessary information from the YAML itself.

```
for file in ${TMPDIR}/*.yaml; do mv $${file} ./pkg/k8s/apis/cilium.io/client/crds/`yq '.spec.versions.0.name' $${file}`/`yq '.spec.names.plural' $${file}`.yaml; done
```

of course this only works as long as a CRD only has one version - which is kind of against the concept of a CRD. but we're "disrespecting" this anyway when trying to move a CRD into a specific version folder.

otherwise we could try to get rid of mapping it to a decicated version folder at all - but i kind of don't have the whole cilium history with CRDs :)

Reply:
I thought about this (using `yq`) but didn't want to involve depending on a new dependency for this refactor. I also thought about an alternative to use some combination of more standard tools like `grep`, but thought that it wasn't really worth the complexity in the end. We don't add new CRDs that often. Additoinally, I think the assumption that a developer already knows what API version their new CRD is going to be under is a safe one to make. Given that, it's just a matter of the developer updating the corresponding variable with their CRD name and then running the target. I didn't see how the additional complexity was beneficial enough for something like this. 

WDYT?

Reply:
definitely agree on using as few external tools as possible in the Makefile - therefore perfectly fine with me!